### PR TITLE
Fix FP S2871 (`no-alphabetical-sort`): Rework the rule to emit a dedicated message for arrays of strings

### LINKED ISSUE #4531: Fix FP S2871 (`no-alphabetical-sort`): Add exception for string arrays
Do not raise the issue when the array is an array of strings:

```js
const names = ['foo', 'bar'];
names.sort(); // compliant
```

### PR DESCRIPTION
**Pull request by ericmorand-sonarsource**
Fixes #4531

### CODE REVIEW

File: packages/jsts/src/rules/S2871/rule.ts [hunk 1]
```diff
@@ -53,6 +53,8 @@ export const rule: Rule.RuleModule = {
     messages: {
       provideCompareFunction:
         'Provide a compare function to avoid sorting elements alphabetically.',
+      provideCompareFunctionForArrayOfStrings:
+        'Provide a compare function that depends on String.localeCompare, to reliably sort elements alphabetically.',
```
Comment (yassin-kammoun-sonarsource) [inline packages/jsts/src/rules/S2871/rule.ts:57]: This looks great!

Comment (yassin-kammoun-sonarsource) [reply packages/jsts/src/rules/S2871/rule.ts:57]:
```suggestion
        'Provide a compare function that depends on "String.localeCompare", to reliably sort elements alphabetically.',
```

File: packages/jsts/src/rules/S2871/unit.test.ts [hunk 2]
```diff
@@ -606,6 +606,8 @@ ruleTester.run(
         code: 'const array = ["foo", "bar"]; const sortedArray = array.toSorted();',
         errors: [
           {
+            message:
+              'Provide a compare function that depends on String.localeCompare, to reliably sort elements alphabetically.',
```
Comment (yassin-kammoun-sonarsource) [inline packages/jsts/src/rules/S2871/unit.test.ts:610]: Might need to apply this suggestion if you applied the one I suggested above.

```suggestion
              'Provide a compare function that depends on "String.localeCompare", to reliably sort elements alphabetically.',
```

File: sonar-plugin/javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S2871.html [hunk 3]
```diff
@@ -31,6 +31,16 @@ <h2>Why is this an issue?</h2>
...
+<h3>Exceptions</h3>
+<p>The rule ignores arrays that only contain strings, or are declared as such.</p>
+<pre>
+const names = ['foo', 'bar'];
+names.sort(); // Compliant
+</pre>
+<pre>
+const names: string[] = [];
+names.sort(); // Compliant
+</pre>
```
Comment (yassin-kammoun-sonarsource) [inline sonar-plugin/javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S2871.html:43]: I guess the exception no longer holds and we can get rid of this from the RSPEC, right?

**Comment by yassin-kammoun-sonarsource** [comment]
We could also update the PR's title since it's no longer about adding an exception for string arrays but rather having a specific message for string arrays.

**Review by ilia-kebets-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**